src/providers/fault-count/fault-count.ts

```diff
@@ -81,7 +102,65 @@ export class FaultCountProvider {
     return 0;
   }
 
-  sumManoeuvreFaults = (manoeuvres: CatBUniqueTypes.Manoeuvres, faultType: CompetencyOutcome): number => {
+  public sumVehicleCheckFaultsCatBE = (vehicleChecks: CatBEUniqueTypes.VehicleChecks): number => {
+    // TODO: Use the Vechicle checks scoring logic provider, to be done in MES-3737
+    return 0;
+  }
+
+  public getDrivingFaultSumCountCatBE = (data: CatBEUniqueTypes.TestData): number => {
+
+    // The way how we store the driving faults differs for certain competencies
+    // Because of this we need to pay extra attention on summing up all of them
+    const { drivingFaults, manoeuvres, vehicleChecks } = data;
+
+    const drivingFaultSumOfSimpleCompetencies =
+      Object.values(drivingFaults).reduce((acc, numberOfFaults) => acc + numberOfFaults, 0);
+
+    const result =
+      drivingFaultSumOfSimpleCompetencies +
+      this.sumManoeuvreFaults(manoeuvres, CompetencyOutcome.DF) +
+      this.sumVehicleCheckFaultsCatBE(vehicleChecks);
+
+    return result;
+  }
+
+  public getSeriousFaultSumCountCatBE = (data: CatBUniqueTypes.TestData): number => {
+
+    // The way how we store serious faults differs for certain competencies
+    // Because of this we need to pay extra attention on summing up all of them
+    const { seriousFaults, manoeuvres, vehicleChecks, eyesightTest } = data;
+
+    const seriousFaultSumOfSimpleCompetencies = Object.keys(pickBy(seriousFaults)).length;
+    const vehicleCheckSeriousFaults = vehicleChecks.showMeQuestion.outcome === CompetencyOutcome.S ? 1 : 0;
```

Comment:
I thought that for B+E tests vehicleChecks.showMeQuestion is actually an array - is that not the case anymore? 
How and when the Serious marker will be checked? Will it be reverted if DE will re-ask the questions again (by going back to that page and amending the results)

Reply:
This will be updated when https://github.com/dvsa/mes-mobile-app/pull/920 goes in..

Reply:
That contains the logic to do this B+E specific calculation for Vehicle Checks, next PR will incorporate that logic in here as well.